packages/serialization/json/src/jsonParseNode.ts

```diff
@@ -78,6 +93,39 @@ export class JsonParseNode implements ParseNode {
     parsableFactory: ParsableFactory<T>,
   ): T => {
     const temp: T = {} as T;
+    if (isUntypedNode(parsableFactory(this)(temp))) {
+      const valueType = typeof this._jsonNode;
+      let value: T = temp;
+      if (valueType === "boolean") {
+        value = createUntypedBoolean(this._jsonNode as boolean) as any as T;
+      } else if (valueType === "string") {
+        value = createUntypedString(this._jsonNode as string) as any as T;
+      } else if (valueType === "number") {
+        value = createUntypedNumber(this._jsonNode as number) as any as T;
+      } else if (Array.isArray(this._jsonNode)) {
+        const nodes: UntypedNode[] = [];
+        // eslint-disable-next-line @typescript-eslint/no-unused-vars
+        (this._jsonNode as any[]).forEach((x) => {
+          nodes.push(
+            new JsonParseNode(x).getObjectValue(
```

Comment:
is there anything that guarantees this is going to be an array of objects instead of an array of strings for example at this stage?

Reply:
That is true. The array is not necessarily an array of objects. 
However, calling `getObjectValue` with the factory as `createUntypedNodeFromDiscriminatorValue` will assure that the type check made by `isUntypedNode`(line 96) on the next recursive call will be true. 
Therefore the array/object will be populated by an UntypedNode type e.g. it will be an array of `UntypedString` rather than `UntypedObject`